Bmx.CommandLine/CommandLine.cs

```diff
@@ -0,0 +1,168 @@
+			throw new NotImplementedException();
+		}
+	}
+
+	public class CommandLine {
+		private BmxCore _bmx;
+		private readonly Parser _cmdLineParser;
+
+		public CommandLine() {
+			// TODO: Fix this :| (ref: Program.cs)
+			_bmx = new BmxCore( new OktaClient( "d2l" ) );
+			_cmdLineParser = BuildCommandLine().UseDefaults().Build();
+
+			_bmx.PromptUserName += GetUser;
+			_bmx.PromptUserPassword += GetPassword;
+			_bmx.PromptMfaType += GetMfaType;
+			_bmx.PromptMfaInput += GetMfaInput;
+			_bmx.PromptRoleSelection += GetRoleType;
+		}
+
+		private CommandLineBuilder BuildCommandLine() {
+			var rootCommand = new RootCommand( "BMX (github.com/Brightspace/bmx), procure AWS credentials" );
+
+			var printCommand = BuildPrintCommand();
+			var writeCommand = BuildWriteCommand();
+
+			rootCommand.Add( printCommand );
+			printCommand.Handler = CommandHandler.Create<string, string, string, string, string>( ExecutePrint );
+
+			rootCommand.Add( writeCommand );
+
+			return new CommandLineBuilder( rootCommand );
+		}
+
+		private Command BuildPrintCommand() {
+			return new Command( "print", "Prints credentials to screen" ) {
+				// TODO: Consider generating these elsewhere, overlaps with write command options
+				new Option<string>( "--account" ) {Required = true, Description = "account name to auth against"},
+				new Option<string>( "--org" ) {Required = true, Description = "okta org api to target"},
+				new Option<string>( "--output" ) {
+					Required = false, Description = "the output format [bash|powershell]"
+				},
+				new Option<string>( "--role" ) {Required = false, Description = "desired role to assume"},
+				new Option<string>( "--user" ) {Required = false, Description = "user to authenticate with"}
+			};
+		}
+
+		private Command BuildWriteCommand() {
+			return new Command( "write" ) {
+				new Option<string>( "--account" ) {Required = true, Description = "account name to auth against"},
+				new Option<string>( "--org" ) {Required = true, Description = "okta org api to target"},
+				new Option<string>( "--output" ) {
+					Required = false, Description = "write to the specified file instead of ~/.aws/credentials"
+				},
+				new Option<string>( "--role" ) {Required = false, Description = "desired role to assume"},
+				new Option<string>( "--user" ) {Required = false, Description = "user to authenticate with"},
+				new Option<string>( "--profile" ) {Required = false, Description = "aws profile name"}
+			};
+		}
+
+		private void ExecutePrint( string account, string org, string role = null, string user = null,
+			string output = null ) {
+			_bmx.Print( account, org, role, user, output );
+		}
+
+		private string GetUser( string provider ) {
+			Console.Write( $"{provider} Username: " );
+			return Console.ReadLine();
+		}
+
+		private string GetPassword( string provider ) {
+			Console.Write( $"{provider} Password: " );
+
+			string pw = "";
```

Comment:
Minor: StringBuilder maybe?

Reply:
String interpolation is a bit more concise so I'd prefer if we use that. String builder makes sense for larger work for performance, although I think nowadays String interpolation is suited for that as well.